scripts/ci.py

```diff
@@ -0,0 +1,133 @@
+import shlex
+
+GIT_ORIGIN_REFIX = "refs/heads/"
+VERBOSE = False
+
+
+def vprint(message):
+    global VERBOSE
+    if VERBOSE:
+        print(message)
+
+
+def running_in_github_action():
+    """
+    Test if currently running in a GitHub action or running locally
+    :return: True if running in GH, False otherwise
+    """
+    return "GITHUB_WORKFLOW" in os.environ
+
+
+def shell(command, cwd=None, check=True):
+    """
+    Run a command
+    :param command: command to run
+    :param shell: Flag indicating if shell should be used by subprocess command
+    """
+    if not shell:
+        command = shlex.split(command)
+
+    vprint(f"running `{command}`")
+    return subprocess.run(command, shell=True, check=check, cwd=cwd, capture_output=True)
+
+
+def get_current_branch(repo_dir):
+    """
+    Get the current branch name for the repository rooted at repo_dir
+    """
+    pout = shell("git branch --show-current", cwd=repo_dir)
+    return pout.stdout.decode("utf-8").strip()
+
+
+def has_remote_branch(repo_dir, branch_name):
+    """
+    Check if the repository rooted at repo_dir has the given branch name. Returns true if it does, false otherwise.
+    """
+    pout = shell(f"git ls-remote --exit-code origin {GIT_ORIGIN_REFIX}{branch_name}", cwd=repo_dir, check=False)
+    return pout.returncode == 0
+
+
+def update_repo(repo_dir, branch_name):
+    """
+    Attempt to update the repository to the given branch name
+    :param repo_dir:
+    :param branch_name:
+    :return:
+    """
+    branch_name = branch_name.removeprefix(GIT_ORIGIN_REFIX)
+    curr_branch = get_current_branch(repo_dir)
+    vprint(f"current branch of `{repo_dir}`: {curr_branch}")
+    if curr_branch == branch_name:
+        vprint("branches match already, nothing to do")
+        return
+
+    has_branch = has_remote_branch(repo_dir, branch_name)
+    if has_branch:
+        vprint(f"repo has target branch `{branch_name}`: {has_branch}...updating")
+        pout = shell(f"git switch {branch_name}", cwd=repo_dir)
+        vprint(pout.stdout.decode("utf-8"))
+    else:
+        vprint(f"repo does not have target branch `{branch_name}`, leaving at {curr_branch}")
+
+
+def _get_branch_cmd(opts):
+    branch = "main"
+    if running_in_github_action():
+        gh_head_ref = os.environ.get("GITHUB_HEAD_REF")
+        gh_ref = os.environ.get("GITHUB_REF")
+        if gh_head_ref:
+            branch = gh_head_ref
+        elif gh_ref:
+            branch = gh_ref.removeprefix(GIT_ORIGIN_REFIX)
+    else:
+        branch = get_current_branch(opts.repository)
+
+    print(branch)
+
+
+def _set_branch_cmd(opts):
+    update_repo(opts.repository, opts.branch)
+
+
+def create_cli():
+    parser = argparse.ArgumentParser(
+        prog="ci",
+        description="Utilities for setting up dependencies correctly in CI",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+
+    parser.add_argument("-v", "--verbose", help="quiet output", action="store_true")
```

Comment:
`--verbose` will make it "quiet output"? or should the help message be "verbose output"

Reply:
whoops, good catch I inverted the flag from quiet to verbose and missed the docs